bundle/config/validate/schema_references.go

```diff
@@ -0,0 +1,130 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+// Validate that any references to UC schemas defined in the DAB use the ${resources.schemas...}
+// syntax to capture the deploy time dependency.
+func SchemaReferences() bundle.ReadOnlyMutator {
+	return &schemaReferences{}
+}
+
+type schemaReferences struct{}
+
+func (v *schemaReferences) Name() string {
+	return "validate:schema_dependency"
+}
+
+func findSchemaInBundle(rb bundle.ReadOnlyBundle, catalogName, schemaName string) ([]dyn.Location, dyn.Path, bool) {
+	for k, s := range rb.Config().Resources.Schemas {
+		if s.CatalogName != catalogName || s.Name != schemaName {
+			continue
+		}
+		return rb.Config().GetLocations("resources.schemas." + k), dyn.NewPath(dyn.Key("resources"), dyn.Key("schemas"), dyn.Key(k)), true
+	}
+	return nil, nil, false
+}
+
+func (v *schemaReferences) Apply(ctx context.Context, rb bundle.ReadOnlyBundle) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	for k, p := range rb.Config().Resources.Pipelines {
+		// Skip if the pipeline uses hive metastore.
+		if p.Catalog == "" {
```

Comment:
The API allows creating a pipeline without a schema/target specified as long as you are not using UC. It's not a forward-looking use case anyway, so I did not look deeper into this.